src/main/java/Duke.java

```diff
@@ -6,70 +7,90 @@ public static void main(String[] args) {
                 System.out.println("Nice! I've marked this task as done:\n " +
-                        storage[num].toString());
+                        storage.get(num).toString());
             }else if(instruction.contains("deadline")){
                 if(instruction.equals("deadline")){
+                    // throws error for incomplete instruction
                     DukeException error = new DukeException(instruction);
                     System.out.println(error.toString());
                 }else {
-                    String[] arr = instruction.split(" /by ");
-                    storage[count] = new Deadline(arr[0].substring(9), arr[1]);
-                    count++;
-                    System.out.println("Got it. I've added this task:\n" + storage[count - 1].toString()
+                    String[] arr = instruction.split(" /by "); // splits the task into description and deadline
+                    storage.add(new Deadline(arr[0].substring(9), arr[1]));
+                    count++; // keep track of number of tasks
+                    System.out.println("Got it. I've added this task:\n" + storage.get(count - 1).toString()
                             + "\n" + "Now you have " + count + " tasks in the list" + ".\n");
                 }
             }else if(instruction.contains("todo")){
                 if(instruction.equals("todo")){
+                    // throws error for incomplete instruction
                     DukeException error = new DukeException(instruction);
                     System.out.println(error.toString());
                 }else {
-                    storage[count] = new Todo(instruction.substring(5));
-                    count++;
-                    System.out.println("Got it. I've added this task:\n" + storage[count - 1].toString()
+                    storage.add(new Todo(instruction.substring(5)));
+                    count++; //keep track of number of tasks
+                    System.out.println("Got it. I've added this task:\n" + storage.get(count - 1).toString()
                             + "\n" + "Now you have " + count + " tasks in the list" + ".\n");
                 }
             }else if(instruction.contains("event")){
                 if(instruction.equals("event")){
+                    // throw error for incomplete instruction
                     DukeException error = new DukeException(instruction);
                     System.out.println(error.toString());
                 }else {
-                    String[] arr = instruction.split(" /from ");
-                    String[] arr1 = arr[1].split(" /to ");
-                    storage[count] = new Event(arr[0].substring(6), arr1[0], arr1[1]);
-                    count++;
-                    System.out.println("Got it. I've added this task:\n" + storage[count - 1].toString()
+                    String[] arr = instruction.split(" /from "); // split task into description and deadline
```

Comment:
Some variable names are not descriptive (like arr and arr1). Maybe more descriptive names are better. (This is hypocritical as I do the same)